tests/attention/test_cake_nvfp4_attention.py

```
@@ -0,0 +1,305 @@
+        f"cosine={raw_cosine:.6f}, mean_abs_error={raw_delta.mean().item():.6f}, "
+        f"max_abs_error={raw_delta.max().item():.6f}"
+    )
+    assert raw_cosine >= cos_threshold, raw_summary
+    assert raw_delta.mean().item() <= mae_threshold, raw_summary
+
+
+def test_cake_nvfp4_attention_rejects_unqualified_large_causal() -> None:
+    """Fail closed on the existing baseline's unqualified S4096 causal row."""
+    _require_cake_device()
+    q = torch.zeros((1, 1, 4096, 128), dtype=torch.bfloat16, device="cuda")
+    with pytest.raises(ValueError, match="causal attention.*seq_len <= 512"):
```

Ruff (0.16.1)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)